transaction.go

```diff
@@ -266,12 +266,12 @@ func (tx *Transaction) matchVariable(match MatchData) {
 	// Array of keys
 	matchedVarsNames := tx.GetCollection(variables.MatchedVarsNames)
 
-	matchedVars.Add("", match.Value)
+	matchedVars.Add(match.Key, match.Value)
```

Comment:
I think ```varName.String()``` should be used instead of ```match.Key``` to prevent confusion of the same key in different ```RuleVariable```